pay_platform_interface/lib/pay_channel.dart

```diff
@@ -34,8 +34,24 @@ import 'pay_platform_interface.dart';
 /// ```
 class PayMethodChannel extends PayPlatform {
   // The channel used to send messages down the native pipe.
-  final MethodChannel _channel =
-      const MethodChannel('plugins.flutter.io/pay_channel');
+  final MethodChannel _channel = const MethodChannel('plugins.flutter.io/pay_channel');
+  static const EventChannel _eventChannel = EventChannel('plugins.flutter.io/pay_events');
+
+  // Stream for payment events
+  final StreamController<Map<String, dynamic>> _eventStreamController = StreamController.broadcast();
+
+  PayMethodChannel() {
+    // Listen to the event channel and add events to the stream controller
+    _eventChannel.receiveBroadcastStream().listen((dynamic event) {
+      final Map<String, dynamic> eventData = jsonDecode(event as String) as Map<String, dynamic>;
+      _eventStreamController.add(eventData);
+    }, onError: (dynamic error) {
+      print('[PayMethodChannel] Error receiving event: $error');
+    });
+  }
+
+  /// Provides a stream of payment events.
+  Stream<Map<String, dynamic>> get events => _eventStreamController.stream;
```

Comment:
Do not forget to use at the app side

Reply:
[91661cf](https://github.com/ekaterina-demeshko-idf/flutter-plugin/pull/2/commits/91661cf33cd80f118bad701885fe6f5c58f3ff4b)